lib/connection.js

```diff
@@ -384,6 +384,8 @@ class Connection extends EventEmitter {
 
     this.server.startSession(this, application, authStrategy, credentials,
       this._onSessionCreated.bind(this));
+
+    this.emit('handshakeRequest', applicationName, authStrategy, credentials);
```

Comment:
Yes because I see no need to, we are not responsible for users breaking their security by themselves.